MapboxDirections/MBWaypoint.swift

```diff
@@ -17,11 +27,13 @@ open class Waypoint: NSObject, NSCopying, NSSecureCoding {
         
         It is recommended that the value of this parameter be greater than the `horizontalAccuracy` property of a `CLLocation` object obtained from a `CLLocationManager` object. There is a high likelihood that the user may be located some distance away from a navigable road, for instance if the user is currently on a driveway or inside a building.
      - parameter name: The name of the waypoint. This parameter does not affect the route but may help you to distinguish one waypoint from another.
+     - parameter allowsArrivingOnOppositeSide: A boolean value indicating whether arriving on opposite side is allowed.
      */
-    @objc public init(coordinate: CLLocationCoordinate2D, coordinateAccuracy: CLLocationAccuracy = -1, name: String? = nil) {
+    @objc public init(coordinate: CLLocationCoordinate2D, coordinateAccuracy: CLLocationAccuracy = -1, name: String? = nil, allowsArrivingOnOppositeSide: Bool = true) {
```

Comment:
Is it necessary to compound the proliferation of initializer arguments? This (for now) relatively obscure feature has a meaningful default, unlike `coordinateAccuracy` or `name`. Waypoint was originally a struct, which is why we put lots of arguments in the initializer, but that isn’t to say that every property needs to have an argument.